### PR TITLE
Replace deprecated / obsolete gethostbyname2() with POSIX.1-2001 getaddrinfo()

### PR DESCRIPTION
**Pull request by michaelortmann**
Found by: michaelortmann
Patch by: michaelortmann
Fixes: 

One-line summary:
Replace deprecated / obsolete gethostbyname2() with POSIX.1-2001 getaddrinfo()

Additional description (if needed):
Please run **misc/runautotools** and **misc/makedepend**

Additionally, this PR will enhance debug logging, example:
`[19:00:56] net: setsockname(): getaddrinfo(): hostname virtual.host.com not known`
with `set vhost4 "virtual.host.com"`

Test cases demonstrating functionality (if applicable):

### CODE REVIEW

File: src/net.c [hunk 1]
```diff
@@ -176,18 +178,28 @@ int setsockname(sockname_t *addr, char *src, int port, int allowres)
...
-      alarm(0);
-    } else
-      hp = NULL;
-    if (hp) {
-      if (hp->h_addrtype == AF_INET)
-        memcpy(&addr->addr.s4.sin_addr, hp->h_addr_list[0], hp->h_length);
+      error = getaddrinfo(src, NULL, NULL, &res0);
+      if (!error) {
+        for (res = res0; res; res = res->ai_next) {
+          if (res == res0 || res->ai_family == (pref_af ? AF_INET6 : AF_INET)) {
```
Comment (thommey) [inline src/net.c:184]: AF_INET6 might be undefined on systems without ipv6 support

**Review by thommey — CHANGES_REQUESTED** [review]

File: src/net.c (hunk 1 above)
Comment (michaelortmann) [reply src/net.c:184]: yes. but i didnt change the visibility of the AF_INET6 usage here, see old code:

https://github.com/eggheads/eggdrop/pull/1348/files#diff-e83dab64ed96e8a50f94260f6a3dea5712ef4150a7c2aa3c1c282d4227c93b66L179

additionally i tested compiling eggdrop with `--disable-ipv6`

**Comment by vanosg** [comment]
@michaelortmann , while we wait for @thommey , can you resolve conflicts? Thanks!

**Comment by vanosg** [comment]
@michaelortmann can you take a look at the compile error we have here?

**Pull request merged**